imports/plugins/included/search-mongo/server/methods/searchcollections.js

```diff
@@ -261,13 +261,48 @@ export function buildOrderSearch(cb) {
   }
 }
 
+export function buildAccountSearchRecord(accountId, updatedFields) {
+  Logger.debug("building account search record");
+  check(accountId, String);
+  check(updatedFields, Array);
+
+  const account = Accounts.findOne(accountId);
+  // let's ignore anonymous accounts
+  if (account && account.emails && account.emails.length) {
+    const accountSearch = {};
+
+    // Not all required fields are used in search
+    // We need to filter through fields that are used,
+    // and only update the search index if one of those fields were updated
+    // forceIndex is included to forceIndexing on startup, or when manually added
+    const searchableFields = ["forceIndex", "emails", "firstName", "lastName", "phone"];
```

Comment:
`requiredFields.accounts` also includes "shopId" so that should be listed here?

Reply:
I don't think shopId should be here.

Based on the [transformations file](https://github.com/reactioncommerce/reaction/blob/fix-3672-kieckhafer-billingShippingAddressChangeBugs/imports/plugins/included/search-mongo/server/methods/transformations.js#L39), it seems that what I've listed here are the only `searchable` fields.

The `required` fields are different - not all are searchable. I'm trying to only update the index when one of the searchable fields is update, not when a required field is updated.

Reply:
OK makes sense

Reply:
Except... if the `shopId` changes in the account, doesn't it also need to change in the `AccountSearch` for security reasons? I'm guessing this is a rare thing, but best to be safe.